### PR TITLE
#9Task. Dzianis Shupenka

### PR DESCRIPTION
**Pull request by dzianisshupenka**
https://travis-ci.org/github/dzianisshupenka/js-assignments/builds/709118318

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -84,6 +94,8 @@ function getPolynom() {
  *   memoizer() => the same random number  (next run, returns the previous cached result)
  */
 function memoize(func) {
+    let result;
+    return () => !result ? result = func() : result;
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can just set the result value and then this ternary expression will not be necessary.

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -171,6 +208,14 @@ function partialUsingArguments(fn) {
...
  */
 function getIdGeneratorFunction(startFrom) {
+    let id = startFrom;
+    return function () {
+        function* generateId(){           
+            while(true)
+              yield id++;
+        }
+        return generateId().next().value
+    }
```
Comment (elizavetaShcherbatykhAis) [inline task/09-functions-n-closures-tasks.js]: You can do it easier :)

**Comment by elizavetaShcherbatykhAis** [comment]
Cool! Please check the comments and try to improve your code
I'll mark the PR as draft please click "ready for review" when it will be finished!
Thank you!

**Comment by dzianisshupenka** [comment]
https://travis-ci.org/github/dzianisshupenka/js-assignments/builds/713047590

**Comment by elizavetaShcherbatykhAis** [comment]
Super! Much better :) I close your PR as accepted

**Pull request closed without merging**